include/swift/AST/PrintOptions.h

```diff
@@ -594,7 +594,7 @@ struct PrintOptions {
   static PrintOptions printDocInterface();
 
   /// Retrieve the set of options suitable for printing SIL functions.
-  static PrintOptions printSIL() {
+  static PrintOptions printSIL(bool printFullConvention = false) {
```

Comment:
Is there something more general we should be passing in here, like a `SILOptions`?

Reply:
We could pass an `Optional<SILPrintContext>` argument and project one of its fields (right now, some callers are doing that), it's not the perfect fit IMO (it carries a bunch of stuff which isn't useful) but we can use it.

Reply:
But `SILPrintContext` lives in SIL, so maybe we don't want to depend on it in the AST.

Reply:
We can declare the method in the AST headers and then implement it in the SIL library somewhere; this wouldn't be the first time we did that.

Reply:
Ok. I already punted some NFC changes to a follow-up PR, I'll address this bit there.